protocol/vm/vm_test.go

```diff
@@ -1,4 +1,4 @@
-package vm
+package vm_test
```

Comment:
Basically I think using actual blocks and transactions here is more of an integration test, and we can probably exercise this package more efficiently (more coverage with less test code) if we focus on unit tests.